HACKING.md

```diff
@@ -0,0 +1,78 @@
+# Tips and tricks for hacking on Buck2
+
```

Comment:
Perhaps a link to contributing here? Since it seems contributing is about "yes, we'd love you to contribute" and Hacking is how to actually do it?